indico/util/user.py

```diff
@@ -88,6 +90,12 @@ def principal_from_fossil(fossil, allow_pending=False, allow_groups=True, allow_
         if group.group is None and not allow_missing_groups:
             raise ValueError('Multipass group does not exist: {}:{}'.format(provider, id_))
         return group
+    elif allow_category_roles and event.category.roles:
+        role = CategoryRole.get(id_)
```

Comment:
You need to handle roles from parent categories here as well (similar to what you do when displaying the list of roles to select from)